packages/shi/src/cli.ts

```diff
@@ -0,0 +1,146 @@
+import { parseArgs } from "util";
+import { operations } from "./operations.js";
+import { shi } from "./shi.js";
+
+// Simple help generation from operations
+const showHelp = () => {
+  console.log("Usage: shi <entity> <operation> [options]\n");
+  console.log("Commands:");
+
+  for (const [entity, ops] of Object.entries(operations)) {
+    for (const [opName, op] of Object.entries(ops)) {
+      console.log(`  ${entity} ${opName} - ${op.description}`);
+      for (const opt of op.options) {
+        console.log(
+          `    -${opt.shorthand}, --${opt.name} <${opt.type}>${opt.required ? " (required)" : ""}`,
+        );
+      }
+      console.log();
+    }
+  }
+
+  console.log("Global options:");
+  console.log("  --provider <n>  LLM provider (openai, anthropic)");
+  console.log("  --model <n>     Model to use");
+  console.log("  -o, --output <file> Write output to file");
+  console.log("  -q, --quiet        Suppress progress messages");
+  console.log("  -h, --help         Show this help");
+  console.log("  -v, --version      Show version");
```

Comment:
It may be more scalable to log these with some mapping function over a predictable data structure of cli options, rather than hard-coding them. But this can be addressed later.